zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java

```diff
@@ -43,20 +45,19 @@ private void updateAndCacheEntity(
       final Record<?> record, final ExportHandler handler, final String id) {
     final var cacheKey = new EntityIdAndEntityType(id, handler.getEntityType());
 
-    final ExporterEntity entity;
-
-    final boolean alreadyCached = cachedEntities.containsKey(cacheKey);
-    if (alreadyCached) {
-      entity = cachedEntities.get(cacheKey).entity();
-    } else {
-      entity = handler.createNewEntity(id);
-    }
+    final EntityAndHandlers entityAndHandlers =
+        cachedEntities.computeIfAbsent(
+            cacheKey,
+            (k) -> {
+              final ExporterEntity entity = handler.createNewEntity(id);
+              return new EntityAndHandlers(entity, new HashSet<>());
```

Comment:
💭 if we care about the order handlers are inserted in we should use `LinkedHashSet` instead of `HashSet`

Reply:
Do we really care? Right now i think it doesnt matter. But happy to be proven otherwise :)

Reply:
right now probably not. but it could happen that in the same batch, there is a handler that add a document, and another that updates it.
`update` operations would fail if the document did not exist

Reply:
But i guess some deterministic behavior wouldnt hurt 👍

Reply:
@mustafadagher I added a new test and the fix, via using the linked hash set.